src/main/kotlin/uk/gov/justice/digital/hmpps/prisonertonomisupdate/services/SynchronisationService.kt

```diff
@@ -49,3 +50,83 @@ data class MappingTelemetry(
   val failureName: String? = null,
   val attributes: Map<String, String> = mapOf(),
 )
+
+class SynchroniseBuilder<MAPPING>(
+  private var transform: suspend () -> MAPPING? = { null },
+  private var fetchMapping: suspend () -> Any? = { null },
+  private var postMapping: suspend (MAPPING) -> Unit = {},
+  var name: String = "unknown",
+  var eventTelemetry: Map<String, String> = mapOf(),
+  var telemetryClient: TelemetryClient? = null,
+  var retryQueueService: RetryQueueService? = null,
+) {
+  suspend fun process() {
+    fetchMapping()?.let {
+      telemetryClient?.trackEvent(
+        "$name-create-duplicate",
+        eventTelemetry,
+        null,
+      )
+    } ?: let {
+      transform().also { mapping ->
+        mapping?.run {
+          val telemetryAttributes = eventTelemetry + mapping.asMap()
```

Comment:
almost certainly adding all mapping fields to telemetry is what we want

Reply:
I'm not sure about this - we pass a lot of data across in the Activities service and most of it is not telemetry. But I'm finding it hard to visualise.

Reply:
This is just the mapping object not the Activity - so just IDs